internal/cmd/operator-sdk/generate/internal/relatedimages.go

```diff
@@ -113,13 +137,15 @@ func (c *relatedImageCollector) collect(name, imageRef, containerRef string) {
 
 func (c *relatedImageCollector) collectedRelatedImages() []operatorsv1alpha1.RelatedImage {
 	final := make([]operatorsv1alpha1.RelatedImage, 0, len(c.relatedImages))
-
+	namesAddedOnce := make(map[string]bool)
```

Comment:
This logic is subject to change based on my previous comments, but you should use `sets.String` from https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets instead of a map when you are just using it to track the existence of a unqiue set of strings.